cli/src/cli_util.rs

```diff
@@ -1228,12 +1257,28 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
         })?;
         drop(progress);
         if new_tree_id != *wc_commit.tree_id() {
+            let update_author_timestamp = {
+                let mut expression = RevsetExpressionEvaluator::new(
```

Comment:
I had to duplicate this logic here to get around the borrow checker. Since `start_working_copy_mutation()` borrows `self` mutably, I can't just call `check_commit_incomplete()`. I also can't seem to move it before locking the workspace, because I don't have `wc_commit` there yet. I'm not sure if there's a better way to do this.